### ISSUE TITLE
Failed Docker image building

### ISSUE THREAD

**Issue by keve60** [IBM/libgroupsig#44]
Hi,

I have some difficulties to build the docker image of the Dockerfile located here `src/wrappers/nodejs/server/Dockerfile`.

After doing this command in `src/wrappers/nodejs/server/` folder
```bash
$ docker build -t groupsig-server .
```

I obtain this error at the stage `npm install`:

```
#12 365.7 -- Installing: /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/trapdoor.h
#12 365.7 -- Installing: /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/proof.h
#12 365.7 -- Installing: /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/klap20.h
#12 365.7 -- Installing: /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml_handles.h
#12 365.8 [ 72%] Completed 'libgroupsig'
#12 365.9 [ 72%] Built target libgroupsig
#12 365.9 Scanning dependencies of target jsgroupsig
#12 365.9 [ 81%] Building C object CMakeFiles/jsgroupsig.dir/src/base64.c.o
#12 366.0 [ 90%] Building C object CMakeFiles/jsgroupsig.dir/src/jsgroupsig.c.o
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:87:22: warning: backslash and newline separated by space
#12 366.1    napi_value jsnull; \
#12 366.1
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_init':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:241:3: error: unknown type name 'groupsig_config_t'; did you mean 'groupsig_proof_t'?
#12 366.1    groupsig_config_t *cfg;
#12 366.1    ^~~~~~~~~~~~~~~~~
#12 366.1    groupsig_proof_t
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:265:10: error: 'groupsig_config_t' undeclared (first use in this function); did you mean 'groupsig_convert'?
#12 366.1    cfg = (groupsig_config_t *) groupsig_init((uint8_t) code, (unsigned int) seed);
#12 366.1           ^~~~~~~~~~~~~~~~~
#12 366.1           groupsig_convert
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:265:10: note: each undeclared identifier is reported only once for each function it appears in
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:265:29: error: expected expression before ')' token
#12 366.1    cfg = (groupsig_config_t *) groupsig_init((uint8_t) code, (unsigned int) seed);
#12 366.1                              ^
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_clear':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:287:3: error: unknown type name 'groupsig_config_t'; did you mean 'groupsig_proof_t'?
#12 366.1    groupsig_config_t *cfg;
#12 366.1    ^~~~~~~~~~~~~~~~~
#12 366.1    groupsig_proof_t
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:313:12: error: too many arguments to function 'groupsig_clear'
#12 366.1    status = groupsig_clear(code, cfg);
#12 366.1             ^~~~~~~~~~~~~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:753:7: note: declared here
#12 366.1    int groupsig_clear(uint8_t code);
#12 366.1        ^~~~~~~~~~~~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_setup':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:450:3: error: unknown type name 'groupsig_config_t'; did you mean 'groupsig_proof_t'?
#12 366.1    groupsig_config_t *cfg;
#12 366.1    ^~~~~~~~~~~~~~~~~
#12 366.1    groupsig_proof_t
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:507:12: error: too many arguments to function 'groupsig_setup'
#12 366.1    status = groupsig_setup((uint8_t) code,
#12 366.1             ^~~~~~~~~~~~~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:773:7: note: declared here
#12 366.1    int groupsig_setup(uint8_t code,
#12 366.1        ^~~~~~~~~~~~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_open':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:977:26: warning: passing argument 1 of 'groupsig_open' from incompatible pointer type [-Wincompatible-pointer-types]
#12 366.1    status = groupsig_open(id, proof, crl, sig, grpkey, mgrkey, gml);
#12 366.1                           ^~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:971:31: note: expected 'uint64_t *' {aka 'long unsigned int *'} but argument is of type 'identity_t *' {aka 'struct <anonymous> *'}
#12 366.1    int groupsig_open(uint64_t *index,
#12 366.1                      ~~~~~~~~~~^~~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_gml_export_file':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:3877:23: warning: passing argument 1 of 'gml_export' from incompatible pointer type [-Wincompatible-pointer-types]
#12 366.1    status = gml_export(gml, dst, format);
#12 366.1                        ^~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:29,
#12 366.1                  from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml.h:248:25: note: expected 'byte_t **' {aka 'unsigned char **'} but argument is of type 'gml_t *' {aka 'struct <anonymous> *'}
#12 366.1  int gml_export(byte_t **bytes, uint32_t *size, gml_t *gml);
#12 366.1                 ~~~~~~~~~^~~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:3877:28: warning: passing argument 2 of 'gml_export' from incompatible pointer type [-Wincompatible-pointer-types]
#12 366.1    status = gml_export(gml, dst, format);
#12 366.1                             ^~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:29,
#12 366.1                  from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml.h:248:42: note: expected 'uint32_t *' {aka 'unsigned int *'} but argument is of type 'char *'
#12 366.1  int gml_export(byte_t **bytes, uint32_t *size, gml_t *gml);
#12 366.1                                 ~~~~~~~~~~^~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:3877:33: warning: passing argument 3 of 'gml_export' makes pointer from integer without a cast [-Wint-conversion]
#12 366.1    status = gml_export(gml, dst, format);
#12 366.1                                  ^~~~~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:29,
#12 366.1                  from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml.h:248:55: note: expected 'gml_t *' {aka 'struct <anonymous> *'} but argument is of type 'uint32_t' {aka 'unsigned int'}
#12 366.1  int gml_export(byte_t **bytes, uint32_t *size, gml_t *gml);
#12 366.1                                                 ~~~~~~~^~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c: In function 'gs_gml_import':
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:3918:37: warning: passing argument 2 of 'gml_import' makes pointer from integer without a cast [-Wint-conversion]
#12 366.1    gml = gml_import((uint8_t ) code, format, src);
#12 366.1                                      ^~~~~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:29,
#12 366.1                  from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml.h:260:8: note: expected 'byte_t *' {aka 'unsigned char *'} but argument is of type 'uint32_t' {aka 'unsigned int'}
#12 366.1  gml_t* gml_import(uint8_t code, byte_t *bytes, uint32_t size);
#12 366.1         ^~~~~~~~~~
#12 366.1 /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:3918:45: warning: passing argument 3 of 'gml_import' makes integer from pointer without a cast [-Wint-conversion]
#12 366.1    gml = gml_import((uint8_t ) code, format, src);
#12 366.1                                              ^~~
#12 366.1 In file included from /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/groupsig.h:29,
#12 366.1                  from /opt/server/node_modules/jsgroupsig/src/jsgroupsig.c:9:
#12 366.1 /opt/server/node_modules/jsgroupsig/build/external/include/groupsig/gml.h:260:8: note: expected 'uint32_t' {aka 'unsigned int'} but argument is of type 'char *'
#12 366.1  gml_t* gml_import(uint8_t code, byte_t *bytes, uint32_t size);
#12 366.1         ^~~~~~~~~~
#12 366.2 make[2]: *** [CMakeFiles/jsgroupsig.dir/build.make:76: CMakeFiles/jsgroupsig.dir/src/jsgroupsig.c.o] Error 1
#12 366.2 make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/jsgroupsig.dir/all] Error 2
#12 366.2 make: *** [Makefile:84: all] Error 2
#12 366.2 ERR! OMG Process terminated: 2
#12 369.4 npm WARN libgroupsig-server@0.0.0 No repository field.
#12 369.4 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
#12 369.4 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
#12 369.4
#12 369.5 npm ERR! code ELIFECYCLE
#12 369.5 npm ERR! errno 1
#12 369.5 npm ERR! jsgroupsig@0.1.4 install: `cmake-js compile`
#12 369.5 npm ERR! Exit status 1
#12 369.5 npm ERR!
#12 369.5 npm ERR! Failed at the jsgroupsig@0.1.4 install script.
#12 369.5 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
#12 369.6
```

If you need more details or information about the error, please let me know. I will do my best to give more relevant information.

Thank you!
Kind regards,
Kévin

Environment: 
```
docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.4
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:55:31 2020
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 18:58:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by jdv-ibm** [comment]
Hi, Kevin.

Thank you very much for pointing this out.

The reason seems to be that I forgot to update the jsgroupsig package version in the `package.json` file. Can you try changing [this line](https://github.com/IBM/libgroupsig/blob/6046944f88f977c7be093e27ab7867a337f4856e/src/wrappers/nodejs/server/package.json#L20) to:

`"jsgroupsig": "^1.0.0",`

And let me know if this solves the issue in your case? (It did solve it for me.)

Best,

Jesus

**Comment by klepretr** [comment]
Jesus,

It's works ! Now, I can build and run the docker image.
Thank you for your quick anwser. 

Kind regards,
Kévin

**Comment by jdv-ibm** [comment]
Thanks!

Fixed in commit f95ea349da73e68f38337a9bcbf4219b46e28456.
